packages/history-service/src/create-root-location-transformer.ts

```diff
@@ -128,6 +137,29 @@ export function createRootLocationTransformer(
       return getConsumerPath(consumerPaths, consumerUid);
     },
 
+    createLocation(
+      ...consumerLocations: ConsumerLocation[]
+    ): history.Location {
+      if (options.primaryConsumerUid && !consumerLocations.some((consumerLocation) => consumerLocation.historyKey===options.primaryConsumerUid)) {
+        throw new Error('primary consumer is mandatory')
+      }
+      let rootlocation: history.Location = {
+        hash:'',
+        pathname:'/',
+        search:'',
+        state:{}
+      };
+      for (const consumerLocation of consumerLocations) {
+        rootlocation = this.createRootLocation(
+          consumerLocation.location,
+          rootlocation,
+          consumerLocation.historyKey
+        ) as history.Location;
```

Comment:
This cast is not correct and obscures potential errors. Shouldn't the return value of the method rather be "LocationDescriptorObject"?

Reply:
yes, we should change ConsumerLocation type. It should contain LocationDescriptorObject. We don't even support hash for any consumer but the primary. I am not sure if we should handle parameters and paths - path is usually enough and could be encoded much easier

Reply:
And what about the key in Location? We cannot properly multiplex that, I guess. It probably doesn't have a meaning for a consumer history. All appearances of Location in the consumerHistory should be cleaned up and a custom Location type without key should be used. I think the key is currently ignored completely.